File: Inverted-Index/src/keywords/Exclude.java
package keywords;
import java.util.HashSet;
import java.util.Set;

import file_handler.Doc;


public class Exclude implements Keyword {

    private String word;

    public Exclude(String word){
        this.word = word;
    }

    public String getWord() {
        return word;
    }
    
    @Override
    public Set<Doc> operate(Set<Doc> docs, Set<Doc> newDocs) {
        if(docs == null) {
            return new HashSet<>();
        }
        if(newDocs != null)
            docs.removeAll(newDocs);
        return docs;
    }
}

File: Phase11-Backend/Models/Keywords/Exclude.cs
using System;
using System.Collections.Generic;
using InvertedIndex.Data.Entities;

namespace InvertedIndex.Models.Keywords
{
    public class Exclude : Keyword
    {
        private string word;

        public Exclude(string word)
        {
            this.word = word;
        }

        public virtual string GetWord()
        {
            return word;
        }
        public virtual HashSet<Doc> Operate(HashSet<Doc> docs, HashSet<Doc> newDocs)
        {
            if(docs == null) {
                return new HashSet<Doc>();
            }
            if(newDocs != null)
                docs.ExceptWith(newDocs);
            return docs;
        }
    }
}